File: qonsult-api/src/main/java/com/qonsult/entity/Question.java
package com.qonsult.entity;

import com.qonsult.enumeration.QuestionTypeEnum;
import lombok.*;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
@Builder
public class Question {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "rank")
    private int rank;

    @Column(name = "label")
    private String label;

    @Column(name = "type")
    @Enumerated(EnumType.STRING)
    private QuestionTypeEnum type;

    @OneToMany(cascade = CascadeType.ALL)
    private List<CodeLabel> content = new ArrayList<>();

    @OneToMany(cascade = CascadeType.ALL)
    private List<Condition> conditions = new ArrayList<>();

}

File: qonsult-fe-patient-portail/src/app/features/dentist-questionnaire/models/question.ts
import { CodeLabel } from "src/app/shared/models/code-label";
import { Condition } from "./condition";
import { QuestionTypeEnum } from "./question-type.enum";

export interface Question {
    id: string;
    label: string;
    type: QuestionTypeEnum;
    rank:number;
    content?: CodeLabel[];
    conditions?: Condition[];
    topic: string;
}
